### DISCUSSION TITLE
Add custome instruction on windows gcc

### DISCUSSION THREAD

**Discussion by roman012285** [xpack-dev-tools/riscv-none-elf-gcc-xpack#25] [General]
I am working with riscv-none-embedd-gcc-xpack I am trying to add a custome instruction to C code or to assembly code. There are few guides in google but they all refer to linux.
Please help

**Comment by ilg-ul** [comment]
1. Building GCC on Windows is a challenge, actually I don't know of any public distributions doing it. All, including riscv-none-embedd-gcc-xpack, cross-compile the Windows binaries on Linux.
2. Unless you are doing this for your personal fun, adding custom instructions to the C compiler is generally not a good idea. At the limit, adding custom instructions to the assembler might be acceptable, but even so generally this approach fragments the RISC-V embedded world and this adds more problems than it solves.

**Reply by roman012285** [reply]
thanks for the reply. I am doing it for research purpose. for example adding instruction avg rd, rs1, rs2 that does average between two registers, it just an example.
Have you done that on Linux, If there is no choice than I will do it on it

**Reply by ilg-ul** [reply]
Ok, good luck with your research.

**Reply by roman012285** [reply]
thanks, have you ever done that on linux?

**Reply by ilg-ul** [reply]
What do you mean by _that_?

**Reply by roman012285** [reply]
sorry I didnt explain myself. that means "adding custome instruction on **Linux**

**Reply by ilg-ul** [reply]
No. Why would I do this? I think that, in principle, if an architecture design encourages users to add custom instructions, this is actually not a feature, but a sign of a poor design, not generic enough. For example I've been using Cortex-M for many years, and I've never felt the need to use anything else in addition to what Arm provided.
